svm/tests/integration_test.rs

```diff
@@ -841,9 +870,105 @@ fn simple_nonce(enable_fee_only_transactions: bool, fee_paying_nonce: bool) -> V
         }
     }
 
+    // 4: safety check that rent-paying nonce fee-payers are verboten (blockhash fee-payers may be below rent-exemption)
+    // if this situation is ever allowed in the future, the nonce account MUST be hidden for fee-only transactions
+    // as an aside, nonce accounts closed by WithdrawNonceAccount are safe because they are ordinary executed transactions
+    // we also dont care whether a non-fee nonce (or any account) pays rent because rent is charged on executed transactions
+    if fee_paying_nonce {
+        let (transaction, _, nonce_info) =
+            mk_nonce_transaction(&mut test_entry, real_program_id, false, true);
+
+        test_entry
+            .final_accounts
+            .get_mut(nonce_info.address())
+            .unwrap()
+            .set_rent_epoch(0);
+
+        test_entry.push_nonce_transaction_with_status(
+            transaction,
+            nonce_info.clone(),
+            ExecutionStatus::Discarded,
+        );
+    }
+
+    // 5: rent-paying nonce fee-payers are also not charged for fee-only transactions
+    if enable_fee_only_transactions && fee_paying_nonce {
+        let (transaction, _, nonce_info) =
+            mk_nonce_transaction(&mut test_entry, Pubkey::new_unique(), false, true);
+
+        test_entry
+            .final_accounts
+            .get_mut(nonce_info.address())
+            .unwrap()
+            .set_rent_epoch(0);
+
+        test_entry.push_nonce_transaction_with_status(
+            transaction,
+            nonce_info.clone(),
+            ExecutionStatus::Discarded,
+        );
+    }
+
     vec![test_entry]
 }
 
+#[allow(unused)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum WriteProgramInstruction {
+    Print,
+    Set,
+    Dealloc,
+    Realloc(usize),
+}
+impl WriteProgramInstruction {
+    fn _create_transaction(
+        self,
+        program_id: Pubkey,
+        fee_payer: &Keypair,
+        target: Pubkey,
+        clamp_data_size: Option<u32>,
+    ) -> Transaction {
+        let instruction_data = match self {
+            Self::Print => vec![0],
+            Self::Set => vec![1],
+            Self::Dealloc => vec![2],
+            Self::Realloc(new_size) => {
+                let mut vec = vec![3];
+                vec.extend_from_slice(&new_size.to_le_bytes());
+                vec
+            }
+        };
+
+        let account_metas = match self {
+            Self::Dealloc => vec![
+                AccountMeta::new(target, false),
+                AccountMeta::new(solana_sdk::incinerator::id(), false),
+            ],
+            Self::Print => vec![AccountMeta::new_readonly(target, false)],
+            _ => vec![AccountMeta::new(target, false)],
+        };
```

Comment:
nit: can we just couple these together so that there's 1 match and it's very clear what the ix_data and account inputs are for each instruction variant?